### DISCUSSION TITLE
Observability: OpenTelemetry

### DISCUSSION THREAD

**Discussion by weyert** [vercel/next.js#41831] [Help] [answered]
### Describe the feature you'd like to request

I woud like to suggest improved support observability through the standard OpenTelemetry which has specifications for tracing, metrics, logging and profiling. Would be beneficial to have support for this without the need to fork or monkey patch.

OpenTelemetry Javascript has GA v1 of Metrics and Tracing, next steps, are working on logging

This was discussed previously in October 2021 with hasn't moved forward

### Describe the solution you'd like

Standardised approach to observe Next application, such as creating metrics (e.g. for API endpoints), and/or tracing. I think it would be awesome if Vercel then would offer an instance of `otel-collector` or OTLP protocol to send traces and logging.

Why OpenTelemetry? It's supported by all major companies in the field, built-in Microsoft's .NET Core, Google, Amazon, NewRelic, Dynatrace etc. 

### Describe alternatives you've considered

Currently, I am experimenting with monkey patching Next 12 to add support for tracing

An example of Opentelemetry with Next that is not an amazing experience can be found at:
https://github.com/open-telemetry/opentelemetry-demo/tree/v1.0.0/src/frontend

**Comment by kharann** [comment]
This would make things like observebility with Grafana easier (Tempo, Loki)

**Comment by danoc** [comment]
There was an update in December saying that it is actively being worked on: https://github.com/vercel/next.js/discussions/32215#discussioncomment-4288693

**Reply by kharann** [reply]
Seems like this is for ejected servers? Would honestly love if it was possible in vercel

**Comment by ScottAwesome** [comment]
@leerob any word on this? Per https://github.com/vercel/next.js/discussions/32215#discussioncomment-4288693 there was something being discussed to this. Would make things a lot either if there was an official Vercel/Next logging package that was isomorphic to every environment.

**Comment by danoc** [comment, marked as answer]
I think this can be closed: https://nextjs.org/docs/app/building-your-application/optimizing/open-telemetry
